### PR TITLE
feat(op-dispute-mon): WithdrawalRequest and Recipients Enricher

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Implements the WithdrawalsRequest Enricher to enrich a game with withdrawal requests from the DelayedWETH contract.
This will be used to validate withdrawal request amounts with credits in the FaultDisputeGame and its associated metrics.

Also implements the Recipients Enricher which refactors recipient list construction out of a bunch of enrichers.
There is probably a bit of cleanup to do across our test suite, but it should be accurate and achieving coverage right now.

### CODE REVIEW

**Comment by refcell** [comment]
* **#9942** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9942?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#9940** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9940?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#9939** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9939?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#9938** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9938?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @refcell and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: op-dispute-mon/mon/extract/withdrawals_enricher.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+func (w *WithdrawalsEnricher) Enrich(ctx context.Context, block rpcblock.Block, _ GameCaller, caller WethCaller, game *monTypes.EnrichedGameData) error {
+	recipients := make(map[common.Address]bool)
+	for _, claim := range game.Claims {
+		if claim.CounteredBy != (common.Address{}) {
+			recipients[claim.CounteredBy] = true
+		} else {
+			recipients[claim.Claimant] = true
+		}
+	}
+	recipientAddrs := maps.Keys(recipients)
```
Comment (ajsutton) [inline op-dispute-mon/mon/extract/withdrawals_enricher.go]: I feel like we've implemented this a number of times in different places.  Maybe we should add it as its own enricher and just store the map of recipients on `EnrichedGameData?  Could be a follow up PR though.

**Review by ajsutton — APPROVED** [review]
Looks good. I'd just adjust to have this get the withdrawals from the `GameCaller` rather than needing to pass `WethCaller` in separately as mentioned in the earlier PR.

File: op-dispute-mon/mon/extract/withdrawals_enricher.go (hunk 1 above)
Comment (refcell) [reply op-dispute-mon/mon/extract/withdrawals_enricher.go]: Very much agree, was planning on doing this in a follow on, but I have time now due to timezone changes so I'll implement in this pr :)

**Pull request merged**